modules/apps/29-fee/module_test.go

```diff
@@ -0,0 +1,281 @@
+			"fee29-1:ics20-1",
+			false,
+			true,
+		},
+		{
+			"valid transfer version on try and counterparty",
+			"ics20-1",
+			"ics20-1",
+			false,
+			true,
+		},
+		{
+			"valid fee middleware and transfer version, crossing hellos",
+			"fee29-1:ics20-1",
+			"fee29-1:ics20-1",
+			true,
+			true,
+		},
+		{
+			"invalid fee middleware version",
+			"otherfee28-1:ics20-1",
+			"fee29-1:ics20-1",
+			false,
+			false,
+		},
+		{
+			"invalid counterparty fee middleware version",
+			"fee29-1:ics20-1",
+			"wrongfee29-1:ics20-1",
+			false,
+			false,
+		},
+		{
+			"invalid transfer version",
+			"fee29-1:wrongics20-1",
+			"fee29-1:ics20-1",
+			false,
+			false,
+		},
+		{
+			"invalid counterparty transfer version",
+			"fee29-1:ics20-1",
+			"fee29-1:wrongics20-1",
+			false,
+			false,
+		},
+		{
+			"transfer version not wrapped",
+			"fee29-1",
+			"fee29-1:ics20-1",
+			false,
+			false,
+		},
+		{
+			"counterparty transfer version not wrapped",
+			"fee29-1:ics20-1",
+			"fee29-1",
+			false,
+			false,
+		},
+		{
+			"fee version not included on try, but included in counterparty",
+			"ics20-1",
+			"fee29-1:ics20-1",
+			false,
+			false,
+		},
+		{
+			"transfer version not included",
+			"fee29-1:ics20-1",
+			"ics20-1",
+			false,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			// reset suite
+			suite.SetupTest()
+			suite.coordinator.SetupClients(suite.path)
+			suite.coordinator.SetupConnections(suite.path)
+			suite.path.EndpointB.ChanOpenInit()
+
+			var (
+				chanCap *capabilitytypes.Capability
+				ok      bool
+				err     error
+			)
+			if tc.capExists {
+				suite.path.EndpointA.ChanOpenInit()
+				chanCap, ok = suite.chainA.GetSimApp().ScopedTransferKeeper.GetCapability(suite.chainA.GetContext(), host.ChannelCapabilityPath(ibctesting.TransferPort, suite.path.EndpointA.ChannelID))
+				suite.Require().True(ok)
+			} else {
+				chanCap, err = suite.chainA.App.GetScopedIBCKeeper().NewCapability(suite.chainA.GetContext(), host.ChannelCapabilityPath(ibctesting.TransferPort, suite.path.EndpointA.ChannelID))
+				suite.Require().NoError(err)
+			}
+
+			suite.path.EndpointA.ChannelID = ibctesting.FirstChannelID
+
+			counterparty := channeltypes.NewCounterparty(suite.path.EndpointB.ChannelConfig.PortID, suite.path.EndpointB.ChannelID)
+			channel := &channeltypes.Channel{
+				State:          channeltypes.INIT,
+				Ordering:       channeltypes.UNORDERED,
+				Counterparty:   counterparty,
+				ConnectionHops: []string{suite.path.EndpointA.ConnectionID},
+				Version:        tc.version,
+			}
+
+			module, _, err := suite.chainA.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainA.GetContext(), ibctesting.TransferPort)
+			suite.Require().NoError(err)
+
+			cbs, ok := suite.chainA.App.GetIBCKeeper().Router.GetRoute(module)
+			suite.Require().True(ok)
+
+			err = cbs.OnChanOpenTry(suite.chainA.GetContext(), channel.Ordering, channel.GetConnectionHops(),
```

Comment:
I think it is possible to call this function directly via `suite.chainA.GetSimApp().FeeKeeper`? (remove looking up module by port)

Reply:
I followed how it was done in the transfer tests.

I think its good to keep anyway because i found routing issues with this flow. This makes sure we can actually grab the right module from the router